docs/concepts/submit-data.md

```diff
@@ -0,0 +1,221 @@
+// keyring from a file and use it to programmatically sign transactions.
+func DemoSubmitData(grpcAddr string, kr keyring.Keyring) error {
+    // create an encoding config that can decode and encode all celestia-app
+    // data structures.
+    ecfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
+
+    // create a connection to the grpc server on the consensus node.
+    conn, err := grpc.Dial(grpcAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+    if err != nil {
+        return err
+    }
+    defer conn.Close()
+
+    // get the address of the account we want to use to sign transactions.
+    rec, err := kr.Key("accountName")
+    if err != nil {
+        return err
+    }
+
+    addr, err := rec.GetAddress()
+    if err != nil {
+        return err
+    }
+
+    // Setup the signer. This function will automatically query the relevant
+    // account information such as sequence (nonce) and account number.
+    signer, err := user.SetupSigner(context.TODO(), kr, conn, addr, ecfg)
+    if err != nil {
+        return err
+    }
+
+    ns := namespace.MustNewV0([]byte("1234567890"))
+
+    fmt.Println("namepace", len(ns.Bytes()))
+
+    blob, err := blobtypes.NewBlob(ns, []byte("some data"), appconsts.ShareVersionZero)
+    if err != nil {
+        return err
+    }
+
+    gasLimit := blobtypes.DefaultEstimateGas([]uint32{uint32(len(blob.Data))})
+
+    options := []user.TxOption{
+        // here we're setting estimating the gas limit from the above estimated
+        // function, and then setting the gas price to 0.1utia per unit of gas.
+        user.SetGasLimitAndFee(gasLimit, 0.1),
+    }
+
+    // this function will submit the transaction and block until a timeout is
+    // reached or the transaction is committed.
+    resp, err := signer.SubmitPayForBlob(context.TODO(), []*tmproto.Blob{blob}, options...)
+    if err != nil {
+        return err
+    }
+
+    // check the response code to see if the transaction was successful.
+    if resp.Code != 0 {
+        // handle code
+        fmt.Println(resp.Code, resp.Codespace, resp.RawLog)
+    }
+
+    // if we don't want to wait for the transaction to be confirmed, we can
+    // manually sign and submit the transaction using the same package.
+    blobTx, err := signer.CreatePayForBlob([]*tmproto.Blob{blob}, options...)
+    if err != nil {
+        return err
+    }
+
+    resp, err = signer.BroadcastTx(context.TODO(), blobTx)
+    if err != nil {
+        return err
+    }
+
+    // check the response code to see if the transaction was successful. Note
+    // that this time we're not waiting for the transaction to be committed.
+    // Therefore the code here is only from the consensus node's mempool.
+    if resp.Code != 0 {
+        // handle code
+        fmt.Println(resp.Code, resp.Codespace, resp.RawLog)
+    }
+
+    return err
+}
+```
+
+### RPC to a celestia-node
+
+> TODO: fill in. Is this the blob module, state module, or rest api? Should we
```

Comment:
since the gateway will soon be fully deprecated, do we only need one section for celestia-node? or two:

1. The celestia-node light client CLI
2. RPC to a celestia-node - this would just be through the JSON RPC, but not the CLI?

Reply:
JSON RPC 😄